src/main/java/com/faforever/client/teammatchmaking/TeamMatchmakingMapListController.java

```diff
@@ -0,0 +1,198 @@
+
+      PlayerInfo player = playerService.getCurrentPlayer();
+      LeaderboardRating ratingBean = player.getLeaderboardRatings().get(queue.getLeaderboard().technicalName());
+      double rating = ratingBean.mean() - 3 * ratingBean.deviation();
+      this.playerBracketIndex = this.getPlayerBracketIndex(this.sortedBrackets, rating);
+
+      List<MapVersion> values = this.sortedBrackets.values().stream().flatMap(List::stream).toList();
+      this.resizeToContent(values.size(), this.TILE_SIZE);
+
+      Flux.fromIterable(values)
+          .publishOn(fxApplicationThreadExecutor.asScheduler())
+          .doOnNext(this::addMapTile)
+          .subscribe();
+    });
+
+  }
+
+
+  private SortedMap<MatchmakerQueueMapPool, List<MapVersion>> getSortedBrackets(Map<MatchmakerQueueMapPool, List<MapVersion>> brackets) {
+
+    Comparator<MapVersion> mapVersionComparator = Comparator.nullsFirst(Comparator.comparing(MapVersion::size))
+                                                            .thenComparing(mapVersion -> mapVersion.map().displayName(),
+                                                                           String.CASE_INSENSITIVE_ORDER);
+
+    Comparator<MatchmakerQueueMapPool> mapPoolComparator = Comparator.comparingDouble(MatchmakerQueueMapPool::minRating)
+                                                                     .thenComparingDouble(
+                                                                         MatchmakerQueueMapPool::maxRating);
+
+    SortedMap<MatchmakerQueueMapPool, List<MapVersion>> sortedMap = new TreeMap<>(mapPoolComparator);
+
+    brackets.forEach((mapPool, mapVersions) -> {
+      List<MapVersion> sortedList = mapVersions.stream().sorted(mapVersionComparator).collect(Collectors.toList());
+      sortedMap.put(mapPool, sortedList);
+    });
+
+    return sortedMap;
+  }
+
+  private SortedMap<MatchmakerQueueMapPool, List<MapVersion>> removeDuplicates(SortedMap<MatchmakerQueueMapPool, List<MapVersion>> sortedBrackets) {
+    HashSet<String> usedMaps = new HashSet<>();
+    sortedBrackets.replaceAll((k, v) -> sortedBrackets.get(k).stream().filter(mapVersion -> {
+      String name = mapVersion.map().displayName();
+      if (usedMaps.contains(name))
+        return false;
+      usedMaps.add(name);
+      return true;
+    }).collect(Collectors.toList()));
+    return sortedBrackets;
+  }
+
+  private Integer getPlayerBracketIndex(SortedMap <MatchmakerQueueMapPool, List<MapVersion>> sortedBrackets, double rating) {
+    int i = 0;
+    for (var entry : sortedBrackets.entrySet()) {
+      MatchmakerQueueMapPool pool = entry.getKey();
+      double min = pool.minRating();
+      double max = pool.maxRating();
+      if (min == 0.0) {min = Double.NEGATIVE_INFINITY;}
+      if (max == 0.0) {max = Double.POSITIVE_INFINITY;}
+
+      if (rating < max && rating > min)
+        return i;
+      i++;
+    }
+    return null;
```

Comment:
If the rating can be null they should be changed to Integer instead of int.

Reply:
to Double, you mean?
I also changed minRating and maxRated types from double to Double in MatchmakerQueueMapPool.java, otherwise no way to distinct null from 0.0